components/camel-aws/src/main/java/org/apache/camel/component/aws/s3/S3Component.java

```diff
@@ -67,8 +67,8 @@ protected Endpoint createEndpoint(String uri, String remaining, Map<String, Obje
         if (ObjectHelper.isEmpty(configuration.getRegion())) {
             setRegion(region);
         }
-        if (configuration.getAmazonS3Client() == null && (configuration.getAccessKey() == null || configuration.getSecretKey() == null)) {
-            throw new IllegalArgumentException("AmazonS3Client or accessKey and secretKey must be specified");
+        if (configuration.isUseIAMCredentials() == false && configuration.getAmazonS3Client() == null && (configuration.getAccessKey() == null || configuration.getSecretKey() == null)) {
+            throw new IllegalArgumentException("useIAMCredentials is set to false, AmazonS3Client or accessKey and secretKey must be specified");
```

Comment:
Since useIAMCredentials is not mandatory, it doesn't make sense to log that it has been set to false.

Reply:
Well, it doesn't make sense with that perspective, but when I start to think about the code functionality, it becomes useful. For the record, I dont think AmazonS3Client is mandatory within the configuration string either and it sort of has the same logic for including isUseIAMCredentials.

Consider the case when a user is expecting to use the "isUseIAMCredentials" option and therefore has not provided any accessKey, secretKey and AmazonS3Client, it would be beneficial to see within the logs (after failure) that the flag has not been set to use the IAMCredentials on the ec2 instance... Its a hint more or less that this is an option that you can consider instead of providing accessKey, secretKey or a custom client (as the only case that you cant provide the other three, you have to provide the iamCredentials). With all of that said, I could care less whether this log makes it into production :)